app/src/main/java/org/p2p/wallet/jupiter/statemanager/SwapStateManager.kt

```diff
@@ -77,14 +77,15 @@ class SwapStateManager(
         when (action) {
             is SwapStateAction.CancelSwapLoading -> return
             is SwapStateAction.TokenAChanged -> {
-                selectedSwapTokenStorage.savedTokenAMint = action.newTokenA.mintAddress
-                if (handleTokenAChange(action.newTokenA)) return
+                handleTokenAChange(action.newTokenA)
+                return
```

Comment:
why you use return here ? It will go out from fun after one of prediction was found

Reply:
Exactly what I want